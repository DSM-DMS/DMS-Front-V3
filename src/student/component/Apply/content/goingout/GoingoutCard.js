import React, { Component } from 'react'
import './GoingoutCard.scss'

const GoingoutCard = ({gooutDate, returnDate, reason}) => {
    const gooutTime = gooutDate.substr(10, 6);
    const returnTime = returnDate.substr(10, 6);
    const date = `${gooutDate[5]}${gooutDate[6]}월 ${gooutDate[8]}${gooutDate[9]}일`;
    
    return (
        <div className = 'apply--goingout--card'>
            <p className = 'apply--goingout--card--title'>외출 신청서</p>
            <p className = 'apply--goingout--card--time'>{`${date} ${gooutTime} ~ ${returnTime}`}</p>
            <p className = 'apply--goingout--card--reason'>{reason}</p>
        </div>
    )
}

export default GoingoutCard;