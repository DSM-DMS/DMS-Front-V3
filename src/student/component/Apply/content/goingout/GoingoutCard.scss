.apply--goingout--card {
    width: 37.5rem;
    background: #fff;
    border-radius: 1rem;
    box-sizing: border-box;
    padding: 1.875rem;
    margin-bottom: 3.125rem;
    box-shadow: 0 5px 10px 0 #29000000;

    &:nth-child(2n-1) {
        margin-right: 6.25rem;
    }

    &--title {
        font-size: 1.5rem;
        font-family: NanumSquareR;
        color: #505050;
        margin: 0;
    }

    &--time {
        margin-top: 1rem;
        font-size: 1.25rem;
        font-family: NanumSquareR;
        color: #606060;
    }

    &--selected {
        background: #19b6b6;
        &>p {
            color: #fff;
        }
    }
}