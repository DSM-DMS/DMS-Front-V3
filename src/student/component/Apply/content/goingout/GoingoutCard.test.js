import React from "react";
import { shallow } from "enzyme";
import GoingoutCard from "./GoingoutCard";

describe('goingout card', () => {
  let component = null;
  const mockOnChangeCard = jest.fn();

  it('renders correctly', () => {
    component = shallow(
      <GoingoutCard
        gooutDate='2019-01-01 08:00'
        returnDate='2019-01-01 09:00'
        reason='아침 식사 외출(서브웨이)'
        selectedClass={undefined}
        onChangeCard={mockOnChangeCard}
        id={123}
      />
    );
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('calls props function', () => {
    const wrapper = component.find('.apply--goingout--card');
    wrapper.simulate('click');
    expect(mockOnChangeCard.mock.calls.length).toBe(1);
  });
});
