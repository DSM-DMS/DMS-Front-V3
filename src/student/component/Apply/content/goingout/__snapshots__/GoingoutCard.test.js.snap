// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goingout card matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoingoutCard
    gooutDate="2019-01-01 08:00"
    id={123}
    onChangeCard={[MockFunction]}
    reason="아침 식사 외출(서브웨이)"
    returnDate="2019-01-01 09:00"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="apply--goingout--card--title"
        >
          외출 신청서
        </p>,
        <p
          className="apply--goingout--card--time"
        >
          01월 01일  08:00 ~  09:00
        </p>,
        <p
          className="apply--goingout--card--reason"
        >
          아침 식사 외출(서브웨이)
        </p>,
      ],
      "className": "unselectable apply--goingout--card undefined",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "외출 신청서",
          "className": "apply--goingout--card--title",
        },
        "ref": null,
        "rendered": "외출 신청서",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "01월 01일  08:00 ~  09:00",
          "className": "apply--goingout--card--time",
        },
        "ref": null,
        "rendered": "01월 01일  08:00 ~  09:00",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "아침 식사 외출(서브웨이)",
          "className": "apply--goingout--card--reason",
        },
        "ref": null,
        "rendered": "아침 식사 외출(서브웨이)",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="apply--goingout--card--title"
          >
            외출 신청서
          </p>,
          <p
            className="apply--goingout--card--time"
          >
            01월 01일  08:00 ~  09:00
          </p>,
          <p
            className="apply--goingout--card--reason"
          >
            아침 식사 외출(서브웨이)
          </p>,
        ],
        "className": "unselectable apply--goingout--card undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "외출 신청서",
            "className": "apply--goingout--card--title",
          },
          "ref": null,
          "rendered": "외출 신청서",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "01월 01일  08:00 ~  09:00",
            "className": "apply--goingout--card--time",
          },
          "ref": null,
          "rendered": "01월 01일  08:00 ~  09:00",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "아침 식사 외출(서브웨이)",
            "className": "apply--goingout--card--reason",
          },
          "ref": null,
          "rendered": "아침 식사 외출(서브웨이)",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
